src/EventListener/CatchAttributesListener.php

```diff
@@ -0,0 +1,76 @@
+<?php
+
+namespace FOS\HttpCacheBundle\EventListener;
+
+use Sensio\Bundle\FrameworkExtraBundle\Configuration\ConfigurationInterface;
+use Symfony\Bundle\FrameworkBundle\Controller\ControllerResolver;
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+use Symfony\Component\HttpKernel\Event\GetResponseEvent;
+use Symfony\Component\HttpKernel\Event\RequestEvent;
+use Symfony\Component\HttpKernel\Kernel;
+use Symfony\Component\HttpKernel\KernelEvents;
+
+if (Kernel::MAJOR_VERSION >= 5) {
+    class_alias(RequestEvent::class, 'FOS\HttpCacheBundle\EventListener\AttributeRequestEvent');
+} else {
+    class_alias(GetResponseEvent::class, 'FOS\HttpCacheBundle\EventListener\AttributeRequestEvent');
+}
+
+class CatchAttributesListener implements EventSubscriberInterface
+{
+    /**
+     * @var ControllerResolver
+     */
+    private $controllerResolver;
+
+    public function __construct(ControllerResolver $controllerResolver)
+    {
+        $this->controllerResolver = $controllerResolver;
+    }
+
+    public function onKernelRequest(AttributeRequestEvent $event)
+    {
+        $request = $event->getRequest();
+
+        if (
+            method_exists(\ReflectionProperty::class, 'getAttributes') &&
```

Comment:
should this listener be loaded at all when we are in php 7? maybe we could instead throw an exception in the constructor (and i would prefer to check the php version over method_exists here as well)